### PR TITLE
BART ce loss ignores pad_token_id instead of -100

### PR DESCRIPTION
**Pull request by sshleifer**
cc @ibeltagy who noticed this.

I think it is better to ignore pad_token_id then -100, but this is semi-breaking because old training code might have replaced pad token id with -100 in labels.

Maybe I should check for both?

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Isn't the default in the config to -100? I think this is fine.

**Comment by sshleifer** [comment]
What do you mean default in config?
The default `ignore_index` is -100 for CrossEntropyLoss.
`pad_token_id` is overwritten by BartConfig.

File: src/transformers/modeling_bart.py [hunk 1]
```diff
@@ -1065,7 +1065,7 @@ def forward(
 
         masked_lm_loss = None
         if labels is not None:
-            loss_fct = CrossEntropyLoss()
+            loss_fct = CrossEntropyLoss(ignore_index=self.config.pad_token_id)
             # TODO(SS): do we need to ignore pad tokens in labels?
```
Comment (sshleifer) [inline src/transformers/modeling_bart.py:1069]:
```suggestion
```

**Comment by sgugger** [comment]
Sorry mixed things up. This would make `BartForConditionalGeneration` behave differently from all the others models (all ModelForMaskedLM and T5ForConditionalGeneration use -100) so I think this is pretty breaking. Users probably have special code to changed padded token to -100, plus you may want to mask other things than the padding for loss computation (more relevant for masked LM than seq2seq but still).

I think this is some preprocessing work to do on the labels, for instance the `DataCollatorForLanguageModeling` replaces all non-masked tokens by -100 in the labels.

**Review by LysandreJik — CHANGES_REQUESTED** [review]
As @sgugger said, let's not have BART behave differently to other models here.
